desc/configuration.py

```diff
@@ -910,704 +885,72 @@ def get_idx_by_xlabel(self, labels):
-            MHD energy (W_B + W_p)
-
-        """
-        R_transform, Z_transform, L_transform = self._get_transforms(grid, derivs=2)
-        pressure = self.pressure.copy()
-        pressure.grid = R_transform.grid
-        iota = self.iota.copy()
-        iota.grid = R_transform.grid
-
-        (
-            energy,
-            magnetic_field,
-            jacobian,
-            cov_basis,
-            toroidal_coords,
-            profiles,
-        ) = compute_energy(
-            self.Psi,
-            self.R_lmn,
-            self.Z_lmn,
-            self.L_lmn,
-            self.p_l,
-            self.i_l,
-            R_transform,
-            Z_transform,
-            L_transform,
-            pressure,
-            iota,
-        )
-
-        return energy
-
-    def compute_quasisymmetry(self, grid=None):
-        """Compute quasisymmetry (triple product and flux function metrics).
-
-        Parameters
-        ----------
-        grid : Grid, ndarray, optional
-            Quadrature grid containing the (rho, theta, zeta) coordinates of
-            the nodes to evaluate at
-
-        Returns
-        -------
-        quasisymmetry: dict
-            dictionary of ndarray, shape(num_nodes,), of quasisymmetry components.
-            The triple product metric has the key 'QS_TP',
-        and the flux function metric has the key 'QS_FF'.
-
-        """
-        R_transform, Z_transform, L_transform = self._get_transforms(grid, derivs=3)
-        pressure = self.pressure.copy()
-        pressure.grid = R_transform.grid
-        iota = self.iota.copy()
-        iota.grid = R_transform.grid
-
-        (
-            quasisymmetry,
-            current_density,
-            magnetic_field,
-            con_basis,
-            jacobian,
-            cov_basis,
-            toroidal_coords,
-            profiles,
-        ) = compute_quasisymmetry(
-            self.Psi,
-            self.R_lmn,
-            self.Z_lmn,
-            self.L_lmn,
-            self.p_l,
-            self.i_l,
-            R_transform,
-            Z_transform,
-            L_transform,
-            pressure,
-            iota,
-        )
-
-        return quasisymmetry
-
-    def compute_volume(self, grid=None):
-        """Compute total plasma volume.
+    # TODO: add kwargs for M_booz, N_booz, etc.
+    def compute(self, name, grid=None, data=None):
+        """Compute the quantity given by name on grid.
 
         Parameters
         ----------
-        grid : Grid, ndarray, optional
-            Quadrature grid containing the (rho, theta, zeta) coordinates of
-            the nodes to evaluate at
+        name : str
+            Name of the quantity to compute.
+        grid : Grid, optional
+            Grid of coordinates to evaluate at. Defaults to the quadrature grid.
 
         Returns
         -------
-        volume : float
-            plasma volume in m^3
+        data : ndarray
```

Comment:
This is not actually what this function returns, at least not in all cases. If I want to compute e^rho, it returns to me a **dict** since that is what the compute_contravariant_basis function does

Reply:
In fact I think nearly every function called from ```compute``` would return a dict, not an ndarray